### ISSUE TITLE
[$60]Forums tab must be displayed for all roles like copilot approver etc

### ISSUE THREAD

**Issue by SathyaJayabal** [topcoder-platform/community-app#4672]
Forums tab must be displayed for all roles like copilot approver etc
example:
https://www.topcoder.com/challenges/30133520
https://beta-community-app.topcoder.com/challenges/0c8061f0-076f-47d3-a901-7cf13f99500c
user: picachui (copilot and appprover for the contest)
<img width="1440" alt="Screenshot 2020-07-22 at 12 50 07 PM" src="https://user-images.githubusercontent.com/58783823/88146989-3c3bf600-cc1a-11ea-9b3d-8cb4441277d0.png">

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30134485 has been created for this ticket.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30134485 has been updated - it has been assigned to cagdas001.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by cagdas001** [comment]
cc @sushilshinde 

**TL;DR;** Needs a backend change at resource-api

The current code uses `challenge.userDetails.roles` to check if user has a role for the challenge.

https://github.com/topcoder-platform/community-app/blob/aabe3f70096a45a9ee1814d9d55ea85dd162861d/src/shared/components/challenge-detail/Header/TabSelector/index.jsx#L37

It fetches user challenges and populates this field from user challenges. (see below code block for more detail)

https://github.com/topcoder-platform/community-app/blob/aabe3f70096a45a9ee1814d9d55ea85dd162861d/src/shared/actions/challenge-listing/index.js#L108-L124

The issue is that, V5 API doesn't return such a field in challenges. I have also checked the V4 API, it also doesn't return. Looks like these are very old codes, were added 1-3 years ago, so I guess `userDetails` is a very old field and is no longer in use.

However, we still need a way to get user's role(s) for a particular challenge. Challenge API doesn't return any helpful info. Looks like there is a function named `getUserRolesInChallenge` in **topcoder-react-lib** repo.

https://github.com/topcoder-platform/topcoder-react-lib/blob/50738814672941b0df4031bfd0713f931dc1bd74/src/services/challenges.js#L692-L711
```js
  /**
   * Gets roles of a user in the specified challenge. The user tested is
   * the owner of authentication token used to instantiate the service.
   *
   * Notice, if you have already loaded the challenge as that user, these roles
   * are attached to the challenge object under `userDetails.roles` path during
   * challenge normalization. However, if you have not, this method is the most
   * efficient way to get them, as it by-passes any unnecessary normalizations
   * of the challenge object.
   *
   * @param {Number} challengeId Challenge ID.
   */
  async getUserRolesInChallenge(challengeId) {
    const user = decodeToken(this.private.tokenV3);
    const url = `/resources?challengeId=${challengeId}?memberHandle=${user.handle}`;
    const resources = await this.private.apiV5.get(url);
    if (resources) return _.map(resources, 'roleId');
    throw new Error(`Failed to fetch user role from challenge #${challengeId}`);
  }
}
```

This is exactly what we're looking for. However, there is an issue with **resource-api**. It only returns the **Submitter** role if the user is not an admin or machine. So, we can't get the **roleId** of other roles like co-pilot etc.

https://github.com/topcoder-platform/resources-api/blob/8637689677040118135c4f184b44812008c5d86f/src/services/ResourceService.js#L58-L64
```js
  if (!currentUser || (!currentUser.isMachine && !helper.hasAdminRole(currentUser))) {
    // await checkAccess(currentUser, resources)
    // if not admin, and not machine, only return submitters
    boolQuery.push({ match_phrase: { roleId: config.SUBMITTER_RESOURCE_ROLE_ID } })
  } else if (roleId) {
    boolQuery.push({ match_phrase: { roleId } })
  }
```

Maybe we can make a change at **resource-api**, if I'm querying another user's resources it's understandable to return only Submitter roles, however, if I'm querying my own resources (this is our case), I think it can return all the resources with all roles. Currently it only returns resources with Submitter roles, even if I'm querying with my own `memberId`

**Comment by sushilshinde** [comment]
@rootelement will confirm that. That's a good analysis @cagdas001

**Comment by cagdas001** [comment]
PR (**topcoder-react-lib**): topcoder-platform/topcoder-react-lib#214

Use this version of **topcoder-react-lib** when testing the **community-app** (you can add it as local dependency)

**Comment by SathyaJayabal** [comment]
verified on beta
<img width="1440" alt="Screenshot 2020-08-05 at 11 12 31 AM" src="https://user-images.githubusercontent.com/58783823/89376040-a2da0d00-d70c-11ea-93c3-3cb736a98c88.png">

**Comment by crazyk07** [comment]
Payment task has been updated: https://software.topcoder.com/review/actions/ViewProjectDetails?pid=30134485<br/><br/>```This is an automated message for crazyk via Topcoder X```
